src/test/e2e/api/admin/public-signup-token.e2e.test.ts

```diff
@@ -0,0 +1,223 @@
+                rootRole: role.id,
+            });
+            req.user = user;
+            next();
+        });
+    };
+
+    const { request, destroy } = await setupAppWithCustomAuth(stores, preHook);
+
+    const tokenCreate: PublicSignupTokenCreateSchema = {
+        name: 'some-name',
+        expiresAt: expireAt().toISOString(),
+    };
+
+    await request
+        .post('/api/admin/invite-link/tokens')
+        .send(tokenCreate)
+        .expect('Content-Type', /json/)
+        .expect(201)
+        .expect((res) => {
+            expect(res.body.name).toBe('some-name');
+            expect(res.body.secret).not.toBeNull();
+            expect(res.body.url).not.toBeNull();
+        });
+
+    await destroy();
+});
+
+test('no permission to validate a token', async () => {
+    const preHook = (app, config, { userService, accessService }) => {
+        app.use('/api/admin/', async (req, res, next) => {
+            const admin = await accessService.getRootRole(RoleName.ADMIN);
+            await userService.createUser({
+                email: 'admin@example.com',
+                username: 'admin@example.com',
+                rootRole: admin.id,
+            });
+            next();
+        });
+    };
+
+    const { request, destroy } = await setupAppWithCustomAuth(stores, preHook);
+
+    await stores.publicSignupTokenStore.insert({
+        name: 'some-name',
+        expiresAt: expireAt(),
+        secret: 'some-secret',
+        createAt: new Date(),
+        createdBy: 'admin@example.com',
+        roleId: 3,
+    });
+    await request
+        .post('/api/admin/invite-link/tokens/some-secret/validate')
+        .expect(200);
+
+    await destroy();
+});
+
+test('should return 401 if token can not be validate', async () => {
+    const preHook = (app, config, { userService, accessService }) => {
+        app.use('/api/admin/', async (req, res, next) => {
+            const admin = await accessService.getRootRole(RoleName.ADMIN);
+            await userService.createUser({
+                email: 'admin@example.com',
+                username: 'admin@example.com',
+                rootRole: admin.id,
+            });
+            next();
+        });
+    };
+
+    const { request, destroy } = await setupAppWithCustomAuth(stores, preHook);
+
+    await request
+        .post('/api/admin/invite-link/tokens/some-invalid-secret/validate')
+        .expect(401);
+
+    await destroy();
+});
+
+test('users can signup with invite-link', async () => {
+    expect.assertions(1);
+
+    const preHook = (app, config, { userService, accessService }) => {
+        app.use('/api/admin/', async (req, res, next) => {
+            const admin = await accessService.getRootRole(RoleName.ADMIN);
+            await userService.createUser({
+                email: 'admin@example.com',
+                username: 'admin@example.com',
+                rootRole: admin.id,
+            });
+            next();
+        });
+    };
```

Comment:
Same question here, seems like this should be this should be validated by the secret

Reply:
It is, the admin user was needed when creating the token